File: backend/Shelter/Domain/Entities/Species.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Domain.Entities
{
    public class Species
    {
        public int Id { get; set; }
        public string Name { get; set; } = default!;

        //Relationships one-to-many
        public ICollection<Animal> Animals { get; set; } = new List<Animal>();

        //Permisions
        public bool RequiresPermit { get; set; }
        public string? PermitName { get; set; }
        public string? PermitAuthority { get; set; }
        public string? PermitNotes { get; set; }

    }
}

File: frontend/Shelter/src/app/core/models/species.ts
export interface SpeciesDto {
  id: number;
  name: string;

  requiresPermit: boolean;
  permitName?: string | null;
  permitAuthority?: string | null;
  permitNotes?: string | null;
}

export interface CreateSpeciesDto {
  name: string;

  requiresPermit: boolean;
  permitName?: string | null;
  permitAuthority?: string | null;
  permitNotes?: string | null;
}